### PR TITLE
Update grpc creds

### PR DESCRIPTION
**Pull request by michaelbausor**
Restructure grpcCredentialsHelpers and add support for passing the Channel and forceNewChannel parameters.

Corresponding toolkit PR: https://github.com/googleapis/toolkit/pull/1454

cc @dwsupplee @jdpedrie

### CODE REVIEW

**Comment by michaelbausor** [comment]
@garrettjonesgoogle @shinfan @neozwu Hi all, gentle ping on this after fixit week

**Comment by shinfan** [comment]
Is there PR fully auto-updated from the toolkit change? Is there any manual update?

**Comment by michaelbausor** [comment]
@shinfan Are you asking if this PR is completely auto-generated? This PR does contain manual changes, but also adds autogenerated files for longrunning (sorry if this is not what you meant)

File: src/GrpcCredentialsHelper.php [hunk 1]
```diff
@@ -42,53 +42,82 @@
...
-        $cachingOptions = array_diff_key($opts, $defaultOptions);
-
-        $this->scopes = $scopes;
+        $args = array_merge($defaultOptions, $args);
 
-        if (isset($opts['credentialsLoader'])) {
-            $credentialsLoader = $opts['credentialsLoader'];
-        } else {
-            $credentialsLoader = $this->getADCCredentials($scopes);
+        if (empty($args['credentialsLoader'])) {
```
Comment (shinfan) [inline src/GrpcCredentialsHelper.php]: What is the behavior if both credentialsLoader and scopes are set?

File: src/GrpcCredentialsHelper.php [hunk 2]
```diff
@@ -99,56 +128,58 @@ public function __construct($scopes, $optionalArgs = [])
...
 
-    // TODO(garrettjones):
-    // add:
-    //   1. (when supported in gRPC) channel
     /**
      * Creates a gRPC client stub.
      *
      * @param callable $generatedCreateStub
-     *        Function callback which must accept two arguments ($hostname, $opts)
+     *        Function callback which must accept two arguments ($hostname, $opts, $channel)
```
Comment (shinfan) [inline src/GrpcCredentialsHelper.php]: s/two/three

Comment (michaelbausor) [inline src/GrpcCredentialsHelper.php]: Done

File: src/GrpcCredentialsHelper.php (hunk 1 above)
Comment (michaelbausor) [inline src/GrpcCredentialsHelper.php]: In this case, `scopes` is ignored (see https://github.com/googleapis/gax-php/pull/91/files/225501ad7904fa26b397e7e4132e730c93239c7e#diff-223c9981f8726481e32c3d2c29471adeR64)

**Pull request merged**